app/jobs/consume_assessments_job.rb

```diff
@@ -138,7 +138,11 @@ def perform
           reported_condition = ReportedCondition.new(symptoms: typed_reported_symptoms, threshold_condition_hash: message['threshold_condition_hash'])
           assessment = Assessment.new(reported_condition: reported_condition, patient: patient, who_reported: 'Monitoree')
           assessment.symptomatic = assessment.symptomatic?
-          queue.commit if assessment.save
+          reported_condition.transaction do
+            assessment.save!
+            reported_condition.save!
+            queue.commit
```

Comment:
@holmesie every other call to `queue.commit` in this file is followed by a call to `next` except these two. Do you know why?

Reply:
Is there any code to even "skip over" by explicitly calling `next` ?

Reply:
Not sure there is any real reason - besides the fact that these two commits are at the very end of potential code paths in this file (barring an error). The loop iteration will complete right after either of these two lines is executed.